docs/designs/elasticdl_operator.md

```diff
@@ -0,0 +1,211 @@
+# Design for ElasticDL Operator
+
+## Motivation
+
+ElasticDL uses master-worker architecture.
+Each ElasticDL job has a unique master pod.
+The master pod manages the lifecycle of worker pods and controls the training process.
+
+ElasticDL provides a command-line [client tool](https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/client_tool.md)
+to submit a job to a Kubernetes cluster.
+At first, a master pod will be launched.
+Then, the master pod launches worker pods and PS pods if necessary.
+The training process begins once a worker pod becomes ready.
+
+When making ElasticDL as a product of cloud computing,
+we find that we have to address the following two points:
+
+- Job monitoring and management. ElasticDL client tool only launches a job.
+We have to write extra scripts to monitor the pods' status
+and clean pods when a job completes.
+
+- Product compatibility. Current products have deployed some operators of [Kubeflow](https://www.kubeflow.org/),
+such as TF operator and PyTorch operator.
+Many development works have been done when integrating the operators,
+including dashboard, command-line tool, and controllers with rich monitoring functions.
+It's better to reuse the work.
+
+So, we decide to apply the
+[operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+to ElasticDL as well.
+We introduce a CRD to define the workload of ElasticDL jobs.
```

Comment:
Give a link for CRD https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/

Reply:
Done